egs/wsj/s5/utils/lang/make_position_dependent_subword_lexicon.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+# 2019 Dongji Gao
+# Apache 2.0.
+
+from make_lexicon_fst import read_lexiconp
+import argparse
+import math
+
+def get_args():
+    parser = argparse.ArgumentParser(description="""This script creates a
+        opsition dependent subword lexicon from a subword lexicon.""")
+    parser.add_argument("--separator", type=str, default="@@", help="""Separator
+        indicate the positon of a subword in word.""")
+    parser.add_argument("lexiconp", type=str, help="""Filename of subword lexicon
+        with pronunciation probabilities, with lines of the form
+        'subword prob p1 p2 ...'""")
+    args = parser.parse_args()
+    return args
+
+def is_end(word, separator):
+    return (not word.endswith(separator))
+
+def write_position_dependent_lexicon(lexiconp, separator):
+    for (word, prob, phones) in lexiconp:
+        phones_length = len(phones)
+        suffix_list = ["_I" for i in range(phones_length)]
```

Comment:
This code pattern is not very easy to read, you should handle all cases directly, not use defaults like this.